vaadin-dev-server/src/main/java/com/vaadin/base/devserver/ThemeModifier.java

```diff
@@ -0,0 +1,304 @@
+            importPresent = true;
+        }
+
+        CSSStyleRule newRule = parseStyleRule(selector, property, value);
+        findRuleBySelector(styleSheet, newRule).ifPresentOrElse(
+                existingRule -> addOrUpdateProperty(existingRule, newRule),
+                () -> styleSheet.addRule(newRule));
+
+        sortStylesheet(styleSheet);
+        writeStylesheet(styleSheet, styles);
+    }
+
+    /**
+     * Removes CSS rule property and removes CSS rule itself if no more
+     * properties are left.
+     */
+    public void removeCssRule(String selector, String property) {
+        File styles = getThemeEditorStyleSheet();
+        CascadingStyleSheet styleSheet = CSSReader.readFromFile(styles,
+                StandardCharsets.UTF_8, ECSSVersion.LATEST);
+
+        if (!importPresent) {
+            insertImportIfNotExists();
+            importPresent = true;
+        }
+
+        // value not considered
+        CSSStyleRule newRule = parseStyleRule(selector, property, "inherit");
+        Optional<CSSStyleRule> optRule = findRuleBySelector(styleSheet,
+                newRule);
+        if (optRule.isPresent()) {
+            CSSStyleRule existingRule = optRule.get();
+            removeProperty(existingRule, newRule);
+            if (existingRule.getDeclarationCount() == 0) {
+                styleSheet.removeRule(existingRule);
+            }
+        }
+
+        writeStylesheet(styleSheet, styles);
+    }
+
+    protected void writeStylesheet(CascadingStyleSheet styleSheet,
+            File styles) {
+        try {
+            CSSWriter writer = new CSSWriter().setWriteHeaderText(true)
+                    .setHeaderText(HEADER_TEXT);
+            writer.getSettings().setOptimizedOutput(false);
+            writer.writeCSS(styleSheet, new FileWriter(styles));
+        } catch (IOException e) {
+            throw new IllegalStateException("Cannot write " + styles.getPath(),
+                    e);
+        }
+    }
+
+    protected void sortStylesheet(CascadingStyleSheet styleSheet) {
+        List<CSSStyleRule> sortedRules = styleSheet.getAllStyleRules().stream()
+                .sorted(Comparator.comparing(
+                        r -> r.getSelectorAtIndex(0).getAsCSSString()))
+                .collect(Collectors.toList());
+        for (CSSStyleRule rule : sortedRules) {
+            List<CSSDeclaration> sortedDeclarations = rule.getAllDeclarations()
+                    .stream()
+                    .sorted(Comparator
+                            .comparing(CSSDeclaration::getAsCSSString))
+                    .collect(Collectors.toList());
+            rule.removeAllDeclarations();
+            sortedDeclarations.forEach(rule::addDeclaration);
+        }
+        sortedRules.forEach(styleSheet::removeRule);
+        sortedRules.forEach(styleSheet::addRule);
+    }
+
+    protected CSSStyleRule parseStyleRule(String selector, String property,
+            String value) {
```

Comment:
```suggestion
    protected CSSStyleRule createStyleRule(String selector, String property,
            String value) {
```

That parsing is used to create the rule is an implementation detail.